### PR TITLE
handleUpgrade wasn't compatible with WS spec

### PR DESCRIPTION
**Pull request by Kamil93**
handleUpgrade wasn't compatible with WS module spec

### CODE REVIEW

File: lib/cws/server.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ export class WebSocketServer extends eventEmitter() {
     if (configs.path && configs.path[0] !== '/') {
       configs.path = `/${configs.path}`;
     }
+    
+    this._upgradeCallback = noop;
```
Comment (goriunov) [inline lib/cws/server.ts]: Can you rename variables to just `upgradeCallback` and move at the top with private tag (it is TypeScript not a JS so it has slightly different approach in creating variables) for reference check out `upgradeListener` variable.

File: lib/cws/server.ts [hunk 2]
```diff
@@ -151,7 +154,7 @@ export class WebSocketServer extends eventEmitter() {
     return socket.end(`HTTP/1.1 ${code} ${name}\r\n\r\n`);
   }
 
-  private handleUpgrade(req: HTTP.IncomingMessage, socket: Socket): void {
+  private handleUpgrade(req: HTTP.IncomingMessage, socket: Socket, upgradeHead, callback): void {
```
Comment (goriunov) [inline lib/cws/server.ts]: You need to specify types for `upgradeHead` and `callback`

File: lib/cws/server.ts [hunk 3]
```diff
@@ -165,6 +168,7 @@ export class WebSocketServer extends eventEmitter() {
         if (!this.serverGroup) return;
 
         this.upgradeReq = req;
+        this._upgradeCallback = callback ? callback : noop;
```
Comment (goriunov) [inline lib/cws/server.ts]: This is probably easier to write `this._upgradeCallback = callback || noop;`

**Comment by Kamil93** [comment]
@goriunov: is it allright now?

File: lib/cws/server.ts [hunk 4]
```diff
@@ -151,7 +153,7 @@ export class WebSocketServer extends eventEmitter() {
     return socket.end(`HTTP/1.1 ${code} ${name}\r\n\r\n`);
   }
 
-  private handleUpgrade(req: HTTP.IncomingMessage, socket: Socket): void {
+  private handleUpgrade(req: HTTP.IncomingMessage, socket: Socket, upgradeHead: any, callback: Function): void {
```
Comment (goriunov) [inline lib/cws/server.ts]: Can you make `upgradeHead: any, callback: Function` optional `upgradeHead?: any, callback?: Function`

Comment (goriunov) [reply lib/cws/server.ts]: If you set up `tslint` in your ide you should be able to see this issues :)

**Pull request closed without merging**